packages/grid/x-grid/src/tests/editRows.XGrid.test.tsx

```diff
@@ -577,4 +577,32 @@ describe('<XGrid /> - Edit Rows', () => {
     expect(cell).to.have.text('1,942');
     expect(apiRef.current.getRow(baselineProps.rows[0].id)!.year).to.equal(1942);
   });
+
+  it('should not allow to save an invalid value with Enter', () => {
+    const { setProps } = render(<TestCase editRowsModel={{}} />);
+    const cell = getCell(1, 0);
+    cell.focus();
+    fireEvent.doubleClick(cell);
+    const input = cell.querySelector('input')!;
+    expect(input.getAttribute('aria-invalid')).to.equal(null);
+    fireEvent.change(input, { target: { value: 'n' } });
+    setProps({ editRowsModel: { 1: { brand: { error: true, value: 'n' } } } });
+    fireEvent.keyDown(input, { key: 'Enter' });
+    expect(input.getAttribute('aria-invalid')).to.equal('true');
```

Comment:
To improve the debug experience?

```suggestion
    expect(input).to.have.attribute('aria-invalid', 'true');
```